atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionLockWatchingCacheView.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.api;
+
+import java.util.Map;
+import java.util.Set;
+
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+
+// todo (gmaretic): decide if the logic for what can be cached should live here or in the cache itself
+public interface TransactionLockWatchingCacheView {
+    /**
+     * Attempts to read the cached versions of cells. If there is no suitable cached value for the cell, it will not be
+     * present in the returned map.
+     *
+     * @param tableRef table to read from
+     * @param cells cells to read
+     * @return a mapping with all cells that were cached and guaranteed to be equal to what the transaction would have
+     * read from the kvs.
```

Comment:
Based on this I figure that a view is created at some point after the transaction is initialised (so you have the start timestamp) and that there is one per transaction - is this right?

Reply:
yup

Reply:
Also, this sentence made no sense so elaborated